packages/vscode-extension/src/manifestListener.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import * as vscode from "vscode";
+import {
+  isDeclarativeCopilotApp,
+  updateIsDeclarativeCopilotApp,
+  workspaceUri,
+} from "./globalVariables";
+import path from "path";
+import {
+  AppPackageFolderName,
+  ManifestTemplateFileName,
+  TeamsAppManifest,
+} from "@microsoft/teamsfx-api";
+import TreeViewManagerInstance from "./treeview/treeViewManager";
+import { ExtTelemetry } from "./telemetry/extTelemetry";
+import { TelemetryEvent, TelemetryProperty } from "./telemetry/extTelemetryEvents";
+import { isValidProjectV3 } from "@microsoft/teamsfx-core";
+
+function setAbortableTimeout(ms: number, signal: any) {
+  return new Promise((resolve, reject) => {
+    const timeoutId = setTimeout(() => {
+      // Resolve the promise after 5 seconds
+      resolve("After timeout. Checking app.");
+    }, ms);
+
+    // Listen for the abort event
+    signal.addEventListener("abort", () => {
+      // Clear the timeout and reject the promise if aborted
+      clearTimeout(timeoutId);
+      reject("resolved after clear");
+    });
+  });
+}
+
+export function manifestListener(): vscode.Disposable {
+  let abortController: undefined | AbortController;
+  const disposable = vscode.workspace.onDidSaveTextDocument(
```

Comment:
AFAIK, this triggers when the manifest file is edited in VS Code. If you want to monitor file changes outside VS Code, you could use `createFileSystemWatcher` to monitor change events.

Reply:
Yes, we only need to track when user edit the file in vscode.